jupyterhub/templates/image-puller/_daemonset-helper.yaml

```
@@ -48,6 +48,9 @@ spec:
       {{- end }}
       tolerations:
         {{- include "jupyterhub.userTolerations" . | nindent 8 }}
+        {{- if .Values.prePuller.extraTolerations  }}
+        {{- .Values.prePuller.extraTolerations  | toYaml | trimSuffix "\n" | nindent 8 }}
+        {{- end }}
```

```suggestion
        {{- with .Values.prePuller.extraTolerations  }}
        {{- .  | toYaml | trimSuffix "\n" | nindent 8 }}
        {{- end }}
```

Helm chart template practices has transitioned to using {{ with "something" }} which only renders if "something" is truthy, such as a non-empty array or string.